File: 2470-removing-stars-from-a-string/2470-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        StringBuilder build = new StringBuilder();
        for (int i = 0; i < s.length(); ++i) {
            if (s.charAt(i) == '*') {
                build.deleteCharAt(build.length() - 1);
            } else {
                build.append(s.charAt(i));
            }
        }
        return build.toString();
    }
}

File: 2470-removing-stars-from-a-string/2470-removing-stars-from-a-string.cpp
class Solution {
public:
    std::string removeStars(std::string s) {
        std::string c;
        for (const auto& el : s) {
            if (el == '*') {
                c.pop_back();
            } else {
                c += el;
            }
        }
        return c;
    }
};